src/i18n.ts

```diff
@@ -0,0 +1,142 @@
+import { registerLocale } from 'react-datepicker';
+import fr from 'date-fns/locale/fr';
+import km from 'date-fns/locale/km';
+import en from 'date-fns/locale/en-US';
+import { translation } from './config';
+import 'moment/locale/km';
+
+export type i18nTranslator = typeof i18n['t'];
+
+export const appResources = {
+  en: {
+    translation: {
+      // Date settings
+      date_locale: 'en',
+      Today: 'Today',
+      about: 'about',
+      'Tap the map to select': 'Tap the map to select',
+      'Click the map to select': 'Click the map to select',
+      Prism: 'Prism',
+      // Main buttons
+      Export: 'Export',
+      Legend: 'Legend',
+      // Menu
+      Hazards: 'Hazards',
+      Vulnerability: 'Vulnerability',
+      Exposure: 'Exposure',
+      Risk: 'Risk',
+      // Analysis
+      'Run Analysis': 'Run Analysis',
+      'Hazard Layer': 'Hazard Layer',
+      'Baseline Layer': 'Baseline Layer',
+      'Choose hazard layer': 'Choose hazard layer',
+      'Choose baseline layer': 'Choose baseline layer',
+      Statistic: 'Statistic',
+      Threshold: 'Threshold',
+      Above: 'Above',
+      Below: 'Below',
+      Mean: 'Mean',
+      Median: 'Median',
+      Date: 'Date',
+    },
+  },
+  // fr: {
+  //   translation: {
+  //     // Date settings
+  //     date_locale: 'fr',
+  //     Today: "Aujourd'hui",
+  //     about: 'Apropos',
+  //     'Click the map to select': 'Cliquez sur la carte pour séléctionner',
+  //     // Main buttons
+  //     Export: 'Exporter',
+  //     Legend: 'Légende',
+  //     // Menu
+  //     Hazards: 'Dangers',
+  //     Vulnerability: 'Vulnerabilités',
+  //     Exposure: 'Exposition',
+  //     Risk: 'Risques',
+  //     // Analysis
+  //     'Run Analysis': 'Lancer une analyse',
+  //     'Hazard Layer': 'Dangers',
+  //     'Baseline Layer': 'Données de référence',
+  //     'Choose hazard layer': 'Choisissez un danger',
+  //     'Choose baseline layer': 'Choisissez des données de référence',
+  //     Statistic: 'Statistique',
+  //     Threshold: 'Limite',
+  //     Above: 'Au dessus de',
+  //     Below: 'En dessous de',
+  //     Mean: 'Moyenne',
+  //     Median: 'Medianne',
+  //   },
+  // },
+};
+
+// Translations are expected to take the form {"fr": {"english sentence": "french translation", ...}
+export const formattedTranslation = Object.keys(translation).reduce(
+  (a, v) => ({ ...a, [v]: { translation: translation[v] } }),
+  {},
+);
+
+const englishKeys = Object.keys(translation)
+  .flatMap(language => {
+    return Object.keys(translation[language]);
+  })
+  .reduce((a, v) => ({ ...a, [v]: v }), {});
+
+export const resources = merge(
+  {
+    en: { translation: englishKeys },
+  },
+  appResources,
+  formattedTranslation,
+);
+
+export const languages = Object.keys(resources);
+
+// Register other date locales to be used by our DatePicker
+registerLocale('en', en);
+registerLocale('fr', fr);
+registerLocale('km', km);
+const moment = extendMoment(Moment as any);
```

Comment:
I remember we've done this before out of neccessity. Is it possible to consolidate it to one util file so we can fix it in one place if we ever find a way to remove the `any`?

Reply:
done